pkg/cmd/cli/secrets/remove_secret_from_object.go

```diff
@@ -0,0 +1,114 @@
+package secrets
+
+import (
+	"fmt"
+	"io"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	RemoveSecretRecommendedName = "remove"
+
+	// TODO: move to examples
+	removeSecretLong = `
+Remove (detach) secrets from a ServiceAccount
+If a secret is no longer valid for a pod, build or image pull, you may remove it from a service account.`
+
+	removeSecretExample = ` // Remove a secret currently associated with a service account like this:
+%[1]s serviceaccount/sa-name secret/secret-name secret/another-secret-name ...`
+)
+
+type RemoveSecretOptions struct {
+	SecretOptions
+}
+
+// NewCmdRemoveSecret creates a command object for detaching one or more secret references from a service account
+func NewCmdRemoveSecret(name, fullName string, f *kcmdutil.Factory, out io.Writer) *cobra.Command {
+	o := &RemoveSecretOptions{SecretOptions{Out: out}}
+
+	cmd := &cobra.Command{
+		Use:     fmt.Sprintf("%s serviceaccount/sa-name secret/secret-name [secret/another-secret-name] ...", name),
```

Comment:
+1, so a couple things to be fixed in terms of consistency and cleanliness: 

* `add|remove` becomes `link|unlink` (or `attach|detach`, `connect|disconnect`, `couple|decouple`)
* remove the need of `kind/` in arguments (unless for any reason we foresee supporting more than one kind for either the service account or the secret(s)).